onnxruntime/core/session/inference_session.h

```diff
@@ -178,12 +178,17 @@ class InferenceSession {
                                           TransformerLevel level = TransformerLevel::Level2) ORT_MUST_USE_RESULT;
 
   /**
-    * Enable a custom set of transformers. Call this before invoking Initialize().
+    * Filter the enabled optimizers (either transformer or rewrite rule) using optimizers_to_disable.
+    * For an optimizer to be enabled, it must be allowed at the current optimization level (as specified in
+    * session options), and NOT in optimizers_to_disable. 
+    * This allows finer grained control of the enabled/disabled optimizations.
+    * Must be called before Initialize() to take effect.
+    * 
     * Calling this API is optional.
-    * When this list is provided ORT ignores the levels set in session options.
     * @return OK if success.
     */
-  common::Status AddCustomTransformerList(const std::vector<std::string>& transformers_to_enable) ORT_MUST_USE_RESULT;
+  common::Status FilterEnabledOptimizers(const std::unordered_set<std::string>& optimizers_to_disable)
```

Comment:
>common::Status [](start = 2, length = 14)

change to void?

Reply:
Considered that but as it's going to be exposed publicly I'd prefer the flexibility to return errors in the future.